### ISSUE TITLE
Changes to the uidmap parameter do not result in actions

### ISSUE THREAD

**Issue by betelgeuse** [containers/ansible-podman-collections#652]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**

1. change uidmap in podman_container  parameters

2. run task on running container

**Describe the results you received:**

"changed": false

**Describe the results you expected:**

"changed": true

**Additional information you deem important (e.g. issue happens only occasionally):**

**Version of the `containers.podman` collection:**
**Either git commit if installed from git: `git show --summary`**
**Or version from `ansible-galaxy` if installed from galaxy: `ansible-galaxy collection list | grep containers.podman`**

```
containers.podman             1.10.3 
```

**Output of `ansible --version`:**

```
ansible [core 2.15.4]
```

**Output of `podman version`:**

```
Version:      3.4.4
API Version:  3.4.4
Go Version:   go1.18.1
Built:        Thu Jan  1 00:00:00 1970
OS/Arch:      linux/amd64
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman/jammy-updates,jammy-security,now 3.4.4+ds1-1ubuntu1.22.04.2 amd64 [installed]
```

**Comment by sshnaidm** [comment]
Please provide a playbook for reproducing, it's not clear from the description what is the problem.

**Comment by betelgeuse** [comment]
```yaml
    - name: Example Ubuntu
      containers.podman.podman_container:
        name: example_ubuntu
        image: "docker.io/ubuntu:22.04"
        command: "tail -f"
        network:
          - host
        uidmap:
          - "0:100000:100000"
```

If you change to this:


```yaml
    - name: Example Ubuntu
      containers.podman.podman_container:
        name: example_ubuntu
        image: "docker.io/ubuntu:22.04"
        command: "tail -f"
        network:
          - host
        uidmap:
          - "0:200000:200000"
```

The container is not recreated


```
TASK [Example Ubuntu] **********************************************************************************************************************************************************************************************************************

ok: [host] => {"actions": [], "changed": false
```

**Comment by sshnaidm** [comment]
I believe it's fixed with #745 . Please reopen if not.
